tools/static_check_containers

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# use github internal registry if available, e.g. within github actions
+if [ "$(curl -s -w "%{http_code}" https://ghcr.io/v2/ -o /dev/null)" = "401" ]; then
+    img=hadolint/hadolint
+else
+    img=ghcr.io/hadolint/hadolint
+fi
+for i in container/*/Dockerfile*; do
+    echo "# Static check of $i"
+    docker run --rm -i $img < $i
```

Comment:
I didn't know [hadolint](https://hadolint.github.io/hadolint/) before. The logo is a cat looking embarassed whilst doing its business. It must be good 😁 I like it